src/components/ReadMore/ReadMore.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useRef, useEffect } from 'react';
+import classnames from 'classnames';
+import { getCommonProps } from '../../utils';
+import Button from '../Button/Button';
+import { ButtonVariants } from '../Button/types';
+import { Collapsible, CollapsibleContent, CollapsibleTrigger } from '../Collapsible';
+export interface ReadMoreProps extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * Unique id for component testing
+   */
+  id?: string;
+  /**
+   * Child element pass in to display as item content.
+   */
+  children?: React.ReactNode;
+  /**
+   * Text for the "Read More" button
+   */
+  readMoreText?: string;
+  /**
+   * Text for the "Read Less" button
+   */
+  readLessText?: string;
+  /**
+   * Maximum height of the content when collapsed
+   */
+  maxHeight?: number;
+}
+
+/**
+ * ## Overview
+ *
+ * A component for displaying expandable content with a "Read More" functionality.
+ *
+ * This component is a wrapper around the Collapsible and Button component and uses the CollapsibleTrigger and CollapsibleContent components to control the expand and collapse functionality.
+ *
+ * [Figma Link](https://www.figma.com/design/hMu9IWH5N3KamJy8tLFdyV/EASEL-Compendium%3A-Tokens%2C-Components-%26-Patterns?node-id=7755-5572&t=JCYbkM8yQcdb51UQ-4)
+ *
+ */
+const ReadMore = ({
+  className,
+  children,
+  readMoreText = 'Read More',
+  readLessText = 'Read Less',
+  maxHeight = 480,
+  ...props
+}: ReadMoreProps) => {
+  const { className: baseClassName, ...commonProps } = getCommonProps(props, 'read-more');
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [hasOverflow, setHasOverflow] = useState(false);
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (contentRef.current) {
+      setHasOverflow(contentRef.current.scrollHeight > maxHeight);
```

Comment:
will it work on Remix? we can link seldon to remix to check that

Reply:
sure, I can check

Reply:
fwiw, attaching a recording of behavior in remix cc @constantinehuzenko @scottdickerson

https://github.com/user-attachments/assets/0e7642eb-990e-4bd5-8bea-59da7ff3911d

you'll notice there is a layout shift below the read more content, which happens because there is a measurement made on page load to determine whether or not to show the button

maybe this is ok, because this content will usually be long and below the fold?

could mitigate by positioning the "Read More" button on top of the gradient and the text, which would be slightly different than the designs

Reply:
this would be the alternative:

https://github.com/user-attachments/assets/1a5618ae-9a11-448d-9c0c-fcdee03bb6d1

this is probably better tbh

vote in the comments below!